examples/tracing_pipeline_runs/compare_traces.py

```diff
@@ -0,0 +1,133 @@
+from dataclasses import dataclass
+from typing import List
+
+import argparse
+import ast
+
+@dataclass
+class Component:
+    name: str
+    type: str
+    n_visits: int
+    input_str: str
+    output_str: str
+    input_parsed: any
+    output_parsed: any
+
+# ToDo: remove when custom tracer is working
+def remove_reset_color(text):
+    return text.replace("\033[0m", "")
+
+def parse_structured_data(data_str: str) -> any:
+    try:
+        # Try to evaluate the string as a Python literal
+        return ast.literal_eval(data_str)
+    except (ValueError, SyntaxError):
+        return data_str
+
+def parse_log_file(file_path: str) -> List[Component]:
+    components = []
+    current_component = {}
+
+    with open(file_path, 'r') as f:
+        for line in f:
+
+            if 'Operation: haystack.component.run' in line:
+                if current_component:
+                    # Parse structured data before creating Component
+                    current_component['input_parsed'] = parse_structured_data(current_component['input_str'])
+                    current_component['output_parsed'] = parse_structured_data(current_component['output_str'])
+                    components.append(Component(**current_component))
+                    current_component = {}
+                continue
+
+            # Parse component properties
+            if 'haystack.component.name=' in line:
+                current_component['name'] = line.split('=')[1].strip()
+
+            elif 'haystack.component.type=' in line:
+                current_component['type'] = line.split('=')[1].strip()
+
+            elif 'haystack.component.visits=' in line:
+                current_component['n_visits'] = int(line.split('=')[1].strip())
+
+            elif 'haystack.component.input=' in line:
+                current_component['input_str'] = line.split('=', 1)[1].strip()
+
+            elif 'haystack.component.output=' in line:
+                current_component['output_str'] = line.split('=', 1)[1].strip()
+
```

Comment:
This can be optimized if needed. WDYT? Something like:

```
HAYSTACK_PREFIXES = {
    'name': 'haystack.component.name=',
    'type': 'haystack.component.type=',
    'visits': 'haystack.component.visits=',
    'input': 'haystack.component.input=',
    'output': 'haystack.component.output='
}
```

```suggestion
            # Parse component properties
             for key, prefix in HAYSTACK_PREFIXES.items():
                if prefix in line:
                    value = line.split('=', 1)[1].strip()
                    if key == 'visits':
                        value = int(value)
                    current_component[f"{'input_str' if key == 'input' else 'output_str' if key == 'output' else key}"] = value
                    break

```